earlydisplay/src/main/java/net/neoforged/fml/earlydisplay/GlDebug.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.fml.earlydisplay;
+
+import net.neoforged.fml.loading.FMLConfig;
+import org.lwjgl.opengl.EXTDebugLabel;
+import org.lwjgl.opengl.EXTDebugMarker;
+import org.lwjgl.opengl.GL32C;
+import org.lwjgl.opengl.GLCapabilities;
+import org.lwjgl.opengl.GLDebugMessageCallback;
+import org.lwjgl.opengl.KHRDebug;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class GlDebug {
+    private static final Logger LOG = LoggerFactory.getLogger(GlDebug.class);
+
+    private GlDebug() {}
+
+    private enum LabelMode {
+        DISABLED,
+        CORE,
+        EXTENSION
+    }
+
+    private enum GroupMode {
+        DISABLED,
+        CORE,
+        EXTENSION
+    }
+
+    private static LabelMode labelMode = LabelMode.DISABLED;
+
+    private static GroupMode groupMode = GroupMode.DISABLED;
+
+    private static int maxLabelLength;
+
+    public static void setCapabilities(GLCapabilities capabilities) {
+        if (!FMLConfig.getBoolConfigValue(FMLConfig.ConfigValue.EARLY_WINDOW_GLDEBUG)) {
+            return;
+        }
+
+        // Setup labeling and markers/groups
+        if (capabilities.GL_KHR_debug) {
+            labelMode = LabelMode.CORE;
+            groupMode = GroupMode.CORE;
+            maxLabelLength = GL32C.glGetInteger(KHRDebug.GL_MAX_LABEL_LENGTH);
+        } else {
+            if (capabilities.GL_EXT_debug_label) {
+                labelMode = LabelMode.EXTENSION;
+                maxLabelLength = 256;
+            }
+            if (capabilities.GL_EXT_debug_marker) {
+                groupMode = GroupMode.EXTENSION;
```

Comment:
Is there any case where `GL_EXT_debug_marker` might be true while `GL_EXT_debug_label` is false, causing `maxLabelLength` to not be set to a non-zero value?